### ISSUE TITLE
language param for google map

### ISSUE THREAD

**Issue by pieterjandebruyne** [doublesecretagency/craft-smartmap#43]
Hi,

I need to translate the google map controls depending on the site locale..

The options that we can pass to smartmaps don't seem to include a language param
https://www.doublesecretagency.com/plugins/smart-map/docs/customizing-the-map-in-twig

The docs on internationalization just change plugin translations or fields
https://www.doublesecretagency.com/plugins/smart-map/docs/internationalization-support 

Not sure if I could use JS but I can't seem to find language params here either
https://www.doublesecretagency.com/plugins/smart-map/docs/manipulating-the-map-in-javascript
https://developers.google.com/maps/documentation/javascript/reference/map

Normally you would pass a language to google maps api like so:
<script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&language=ja&region=JP">
</script>

Any way to pass this using smartmaps?

Thanks!

VERSIONS:
Craft 2.7.4
Smartmaps 2.4.1

**Comment by lindseydiloreto** [comment]
Hi @pieter-janDB,

Before answering your question, let me recommend that you update to the latest version of Smart Map (v2.5.1) for Craft 2. It doesn't address your issue, but adds & fixes several important things...

https://www.doublesecretagency.com/plugins/smart-map/changelog#2-5-1

Back to your question... There's no easy way to do this, unfortunately. If you know that your site _only_ needs to show results in Japanese, then you can hack the plugin to render the map in Japanese. Otherwise, it will be fairly difficult to change this.

It's a great feature request though. I'll keep a note of it here for future reference.

**Comment by BenParizek** [comment]
I was hoping to do this too. I'm not familiar enough with the plugin or Google Maps API to offer any suggestions (at the moment) but it does look like it's possible with the Google Maps API:

https://developers.google.com/maps/documentation/javascript/localization
https://developers.google.com/maps/documentation/javascript/examples/map-language#maps_map_language-html

My ideal implementation would look something like this in Twig, or at least let us manually map and pass a language to the options array:

```
{{ craft.smartMap.map(locations, {
  language: currentSite.language
}) }}
```

**Comment by lindseydiloreto** [comment]
Good news, this is possible (and easy) with an upcoming major release! It'll probably be released within the next 4-6 weeks, and/or shortly after Craft 3.6 is released (it requires a minimum of 3.6).

In the meantime... I recently helped someone put together a workaround for this exact issue. Here's what we came up with:

```twig
{% set smartMapJs %}
    {{ craft.smartMap.js }}
{% endset %}

{% set browserKey = craft.smartMap.browserKey %}
{% set keyAdditions = browserKey ~ '&region=JP&language=ja' %}
{% set smartMapJs = smartMapJs|replace(browserKey, keyAdditions) %}

{{ smartMapJs|raw }}
```

In summary, you are pre-loading the Smart Map (and Google Maps) JavaScript. But instead of immediately outputting them, you are editing the Google Maps URL.

By determining your `browserKey`, you have a convenient place to append whatever query parameters you want to the API URL.

Make sure that the `{{ smartMapJs|raw }}` is put in a reasonable place. This would naturally appear in the footer, but it is up to you when you want these scripts to be called.

Hope that helps!

**Comment by lindseydiloreto** [comment]
Good news, this is done in the new Google Maps plugin! It's now super easy to change the language of a map.

 - [Changing the Map Language](https://plugins.doublesecretagency.com/google-maps/guides/changing-map-language/)

```twig
{{ googleMaps.map(locations).tag({
    'api': {
        'language': 'ja',
        'region': 'JP'
    }
}) }}
```
